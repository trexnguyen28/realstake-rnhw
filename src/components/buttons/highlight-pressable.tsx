import React from 'react';
import {Pressable, StyleSheet, Text} from 'react-native';
import {fontStyles} from '@themes';
import {useAppTheme} from '@hooks';
//
import {RsPressableProps} from './rs-pressable';

const styles = StyleSheet.create({
  title: {
    ...fontStyles.Body,
    textDecorationLine: 'underline',
  },
});

interface HighlightPressableProps extends RsPressableProps {
  title?: string;
}

const HighlightPressable: React.FC<HighlightPressableProps> = ({
  title,
  ...restProps
}) => {
  const {colors} = useAppTheme();

  return (
    <Pressable {...restProps}>
      <Text style={[styles.title, {color: colors.highlight}]}>
        {title || '-'}
      </Text>
    </Pressable>
  );
};

export {HighlightPressable};
