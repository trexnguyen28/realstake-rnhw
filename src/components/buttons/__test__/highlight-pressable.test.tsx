import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';
//
import {HighlightPressable} from '../highlight-pressable';

describe('Highlight Pressable render', () => {
  it('render title correctly', () => {
    const tree = renderer
      .create(
        <HighlightPressable title={'Test 1'}>
          <View style={{backgroundColor: 'green'}} />
        </HighlightPressable>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render title and event handler correctly', () => {
    const tree = renderer
      .create(
        <HighlightPressable
          title={'Test 2'}
          onPress={() => console.log('Test 2')}
        >
          <View style={{backgroundColor: 'red'}} />
        </HighlightPressable>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
