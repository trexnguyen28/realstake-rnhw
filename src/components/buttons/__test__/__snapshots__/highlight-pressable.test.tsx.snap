// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight Pressable render render title and event handler correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "letterSpacing": 0.3,
          "lineHeight": 20,
          "textDecorationLine": "underline",
        },
        Object {
          "color": "#0094FF",
        },
      ]
    }
  >
    Test 2
  </Text>
</View>
`;

exports[`Highlight Pressable render render title correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "letterSpacing": 0.3,
          "lineHeight": 20,
          "textDecorationLine": "underline",
        },
        Object {
          "color": "#0094FF",
        },
      ]
    }
  >
    Test 1
  </Text>
</View>
`;
